### DISCUSSION TITLE
Help Understand dataset = Planetoid(path, args.dataset, transform=T.NormalizeFeatures())

### DISCUSSION THREAD

**Discussion by utkarsh0902311047** [pyg-team/pytorch_geometric#7585] [Q&A]
I am trying to run gcn.py example and I see the line "dataset = Planetoid(path, args.dataset, transform=T.NormalizeFeatures())". In this line can anyone tell me how to observe the effects of NormalizeFeatures. Because I am not able to observe any difference in the dataset which gets download with or without NormalizeFeatures.

Thank you

**Comment by rusty1s** [comment]
The `NormalizeFeatures` will normalize the node features row-wise such they get summed up to one. It only has an effect if you directly access the data inside the dataset, i.e. via `data = dataset[0]`.

**Reply by utkarsh0902311047** [reply]
thank you for your response. 
Therefore how can I take the transformed Cora dataset files and convert them to .csv format so that can export them to another tool?

**Reply by rusty1s** [reply]
What do you wanna do exactly? You can create a `pandas.DataFrame` both for node features and edge indices and save them to CSV, e.g.:
```python
x_df = pd.DataFrame({i: data.x[:, i] for i in range(data.x.size(1))}
edge_index_df = pd.DataFrame({
    'src': data.edge_index[0],
    'dst': data.edge_index[1],
})
